pkg/lockfile/parse.go

```diff
@@ -126,11 +126,19 @@ func Parse(pathToLockfile string, parseAs string) (Lockfile, error) {
 	parser, parsedAs := FindParser(pathToLockfile, parseAs)
 
 	if parser == nil {
+		if parseAs != "" {
+			pathToLockfile += fmt.Sprintf(" (parsed as %s)", parseAs)
```

Comment:
Is there a strong reason to add this to the path here? 

This seems inconsistent and would break the interface for users expecting `Lockfile.FilePath` to be a proper filepath. If this is for printing purposes, we can always add this in later?

Reply:
This is assigning to the function parameter which never gets mapped to `Lockfile.FilePath` in this branch because at this point we're _always_ going to return an empty `Lockfile` + error - so this doesn't break anything :)

Reply:
Got it, thanks. I got confused with this variable name being re-used. 

In that case, I think we should return something more specific to that case.

i.e.

```go
if parseAs != "" {
  return Lockfile{}, fmt.Errorf("Requested parser for %s not found", parseAs)
}
```

Or something similar. 

`var ErrParserNotFound = errors.New("could not determine parser")` also seems a bit misleading for this particular case, and the full error message seems a bit confusing:

"could not determine parser for foo/bar (parsed as blah)"

Reply:
> var ErrParserNotFound = errors.New("could not determine parser") also seems a bit misleading for this particular case,

Do you mean the error or the message? as I think the error is accurate but agree the message is not optimal for this case - overall though I think it is more important this returns `ErrParserNotFound` so that you can use `errors.Is` to properly test for it.

Reply:
Yep my main concern was just the message itself.